### PR TITLE
 #162171018 user password reset

### PR DESCRIPTION
**Pull request by akpante3**
#### What does this PR do?
It Resets users password

#### Description of Task to be completed?
1. made use of nodemailer to send mails to user
2. created 2 routes /forgotpassword and /resetpassword/:token
3. reset password 
4. updated branch with ft-signup-validation branch 
5. wrote test for endpoints and validations
6. added a before block in my test
7. changed route/Users.js to route/user.js
8. changed from the use of nnode mailer to sendGrid

#### How should this be manually tested?
on your console
- npm start

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?

[#162171018](https://www.pivotaltracker.com/story/show/162171018)

#### Screenshot
N/A
#### Questions:
N/A

### CODE REVIEW

File: test/users.js [hunk 1]
```diff
@@ -1,15 +1,173 @@
...
+            confirmPassword: '123456'
+          });
+        if (res) {
+          res.status.should.equal(401);
+          res.body.should.be.a('object');
+          res.body.should.have.property('message');
+          res.body.message.should.equal('error decoding token');
+        }
+      } catch (error) {
+        logger.info(error);
```
Comment (thebere) [inline test/users.js]: remove all logger.info from test files

File: index.js [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 const app = express();
 
 app.use(morgan('dev'));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
+app.use('/v1/api/', userRoutes);
 
 app.listen(PORT, () => {
-  logger.info(`connected on port ${PORT}`);
+  console.log(`connected on port ${PORT}`);
```
[the remark this answers is not in the document]
Comment (akpante3) [reply index.js]: done

File: test/users.js (hunk 1 above)
Comment (akpante3) [reply test/users.js]: done

**Comment by azupatrick0** [comment]
Nice work @akpante3

**Pull request merged**